### DISCUSSION TITLE
Ready made docker image for CI

### DISCUSSION THREAD

**Discussion by eadf** [fltk-rs/fltk-rs#1489] [Q&A]
Does anyone know of a ready made [docker image](hub.docker.com) that already contains the required packages for rust + fltk?
Using something like this would speed up the CI process a lot.

I usually have something like this is my CI scripts, using the "normal" rust image:
```sh
apt-get install -y build-essential cmake libpango1.0-dev libx11-dev libxext-dev libxft-dev libxinerama-dev libxcursor-dev libxrender-dev libxfixes-dev
```
 And that can take a while to process.
 
 Edit: I use the woodpecker CI [like this](https://ci.codeberg.org/repos/12850/pipeline/2/config)

**Comment by ruanformigoni** [comment]
Hi! You can create and upload your own image to [dockerhub](https://docs.docker.com/guides/walkthroughs/publish-your-image/), if you want something more lightweight, you could use alpine, [here](https://github.com/ruanformigoni/gameimage/blob/master/deploy/Dockerfile.alpine.wizard) is an example of an environment with rust and fltk on it. Here is a simplified example (you might not need all these packages though):

```dockerfile
FROM alpine:latest

# Switch to edge
RUN echo https://dl-cdn.alpinelinux.org/alpine/edge/main/ > /etc/apk/repositories
RUN echo https://dl-cdn.alpinelinux.org/alpine/edge/community/ >> /etc/apk/repositories
RUN echo https://dl-cdn.alpinelinux.org/alpine/edge/testing/ >> /etc/apk/repositories

# Install packages
RUN apk update
RUN apk add git wget curl patchelf py3-pip pipx file build-base cmake \
  fuse3-dev libxinerama-dev libxcursor-dev libxfixes-dev libxft-dev pango \
  pango-dev libayatana-appindicator libayatana-appindicator-dev gtk+3.0-dev \
  bash fuse fuse3 zlib-static glib-static cairo-static lddtree font-noto

# Symlink compilers
RUN ln -sfT /usr/bin/gcc /usr/bin/musl-gcc
RUN ln -sfT /usr/bin/g++ /usr/bin/musl-g++

# Compile as shared libraries
ENV RUSTFLAGS='-C target-feature=-crt-static'

# Install rust
RUN curl https://sh.rustup.rs -sSf | sh -s -- -y

# Update env
ENV PATH="/root/.cargo/bin:$PATH"
```

**Reply by eadf** [reply]
Thanks, but I'm kinda lazy so I'd prefer a maintenance free option, one that will be updated for each rust release.

**Reply by ruanformigoni** [reply]
I see, you can setup a github-ci job that re-builds the image between a specified period (7 days for example). Then use docker pull on it.
